### ISSUE TITLE
[Bug] Shell Application crash when clicking hardware back button on android

### ISSUE THREAD

**Issue by ColeXm** [xamarin/Xamarin.Forms#7327]
### Description

Shell Application crash when clicking hardware back button on android (back to device homepage), the error shows `System.NullReferenceException`, but it occurs on latest Xamarin.Forms (4.2.0.709249) , i test with the version `4.1.0.555618` and  no crash happens , i think it's a bug. 

### Steps to Reproduce

1. Download official sample :https://docs.microsoft.com/en-us/samples/xamarin/xamarin-forms-samples/userinterface-xaminals/ .
2. Upload Xamarin.Forms to the latest .
3. Test on android,  when locates at root page , click hardware back button .

### Expected Behavior

Back to HomePage of mobile phone.

### Actual Behavior

Crash occurs.

### Basic Information

- IDE:  Visual studio 2019 16.1.3
- Xamarin.Android : 9.3.0.22 
- Nuget Packages: Xamarin.Forms (4.2.0.709249)

### Screenshots

![33](https://user-images.githubusercontent.com/2161256/64029722-c3522c80-cb77-11e9-8144-701af9975b1a.gif)

**Comment by BrunoLogan** [comment]
I'm having the same issue but on 4.1.0.709244

**Comment by chrisparkeronline** [comment]
Just adding a tad bit to this. I get "System.NullReferenceException: 'Object reference not set to an instance of an object.'" when using the hardware back button.

**Comment by Sauder91** [comment]
Probably already seen something like this but this is the stack trace I'm getting.
<img width="1360" alt="image" src="https://user-images.githubusercontent.com/37815396/64668174-3a15e200-d422-11e9-85e1-0b07ad34fe81.png">

**Comment by adrianknight89** [comment]
This is a duplicate of #6640.

#6715 fixes that.

**Comment by polieadie** [comment]
I was using version 4.2.0.815419 and I was having the same issue.
I downgrade to 4.1.0.555618 as mentioned by @raevenbauto then by @ColeXm and i don't have the issue anymore.

**Comment by adamsproul** [comment]
Same experience as @polieadie & @raevenbauto>

I had to downgrade Xamarin.Forms.Maps before Xamarin.Forms in order to downgrade successfully.

4.2.0.815419 --> 4.1.0.555618

**Comment by mryll** [comment]
I have the same problem.

Xamarin.Forms: 4.2.0.815419

I had to disable the back button so it doesn't go to the home page.

**Comment by pengalo** [comment]
Microsoft, have you reproduced this issue? This is now a blocker for me because the previous versions have a bug with custom shell title (Shell.TitleView) where it obscures the top of the main page content. I can't release with either bug.

**Comment by PureWeen** [comment]
@pengalo fixed by

https://github.com/xamarin/Xamarin.Forms/pull/7768

**Comment by polieadie** [comment]
I just tried with version 4.2.0.848062 (latest).
Still the same problem with back button.
I have no choice but to go back to 4.1.0.555618.

Do we have a timeline when this will be solved on a released version?

**Comment by pichow** [comment]
Same here with 4.2.0.709249

**Comment by gonzalesgm-git** [comment]
its still a problem in 4.4.0.991265

**Comment by kvpt** [comment]
Can someone provide a basic reproduction, because the reproduction case provided in #6640 seem fixed.

From the stack trace in https://github.com/xamarin/Xamarin.Forms/issues/7327#issuecomment-530214066, the issue seem triggered by the fragment destroy.

In my PR (#6715) I replaced all destroy with dispose, so I'm almost sure that it will fix the issue.
I rebased the changes on top of the 4.4 branch.
If someone can test that, need to rename the zip to nupkg.
[Xamarin.Forms.4.4.99.zip](https://github.com/xamarin/Xamarin.Forms/files/3965535/Xamarin.Forms.4.4.99.zip)

If it work I will resurrect this PR.
